### PR TITLE
feat(status-bar): rich PR display with state icon, checks dot, tooltip

### PR DESCRIPTION
**Pull request by DaniAkash**
## Summary

- Removes the bare \`#<num>\` chip from the sidebar tab row and replaces it with a rich PR pill in the status bar (between the git segment and the cwd segment).
- Pill renders: state icon (open / draft / merged / closed), \`#<num> + truncated title\`, a CI checks dot (green / yellow / red), and a tooltip with full title, state label, ahead/behind echo, and a 4-bucket check breakdown.
- Click on the pill opens the PR in the OS default browser (\`plugin-opener\`).

## How PR state stays fresh

\`GitMonitorMod\` gains an adaptive refresh cadence:

- **60s baseline** when no PR checks are pending (or when there's no PR).
- **15s** while \`checks.pending > 0\` so CI transitions surface quickly during active runs.
- The mod re-derives the desired interval after every query and the periodic timer wakes early on cadence changes via \`tokio::watch\` — no new polling task per tab, no extra \`gh\` calls when CI is idle.
- The existing OSC 133;D trigger keeps things sub-second after any shell command that could change PR state (push, commit, \`gh pr ready\`, etc.).

\`gh pr view --json\` now also requests \`isDraft\`, \`mergedAt\`, and \`statusCheckRollup\`. The rollup is collapsed on the Rust side into a fixed 4-bucket counter so the frontend treats it as a closed type — unknown check states from future \`gh\` versions are dropped, never miscategorised. \`mergedAt\` disambiguates older \`gh\` versions that return \`state: \"CLOSED\"\` for merged PRs.

## Test plan

- [x] \`bunx tsc --noEmit\` clean
- [x] \`bunx biome check\` clean on changed files
- [x] \`bun test src/components/StatusBar/PrItem.test.ts\` — 16/16 pass (truncate / stateLabel / checksColor coverage including skipped-as-success, MERGED-overrides-isDraft, dot precedence)
- [ ] \`cargo test --manifest-path src-tauri/Cargo.toml --lib git_monitor\` — covers \`transform_pr\` / \`summarise_checks\` / \`desired_interval_secs\` (open + checks bucketing, CLOSED+mergedAt → MERGED, CLOSED without mergedAt stays CLOSED, MERGED passthrough, empty rollup, external \`state\` field, draft flag, missing isDraft default, unknown future state dropped, cadence selection)
- [ ] Manual QA: tab on a branch with open PR shows pill; click opens browser; hover shows tooltip; \`gh pr ready\` on a draft flips icon within ~60s (or immediately after any shell command); pushing a failing commit turns dot red within 15s; merging on github.com flips icon to \`GitMerge\`; tab on branch without PR hides the pill; tab on PR-but-no-CI shows pill with no dot and \"no checks reported\" in the tooltip
- [ ] Window-resize sanity — long titles clip via existing \`overflow-hidden\`

### CODE REVIEW

File: src/components/StatusBar/PrItem.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+              className="inline-block h-1.5 w-1.5 shrink-0 rounded-full"
+              style={{ background: dotColor }}
+              data-testid="pr-checks-dot"
+            />
+          )}
+          <StateIcon pr={pr} />
+          <span style={{ color: stateColor(pr) }}>#{pr.number}</span>
+          <span className="truncate opacity-80">
+            {truncate(pr.title, TITLE_MAX)}
+          </span>
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src/components/StatusBar/PrItem.tsx:235]: Acknowledged — the JS-side `truncate(pr.title, 40)` already caps the title before render, so the CSS `truncate` class is effectively dead defence-in-depth. Leaving as-is for now; happy to drop the class in a follow-up cleanup.

File: src/components/StatusBar/PrItem.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+ * adding a separate polling task. The OSC 133;D trigger keeps things
+ * sub-second after any shell command that could change PR state (push,
+ * commit, force-push, gh pr ready, etc.).
+ * -------------------------------------------------------------------------*/
+
+/** Max chars for the inline title before truncating with an ellipsis. */
+export const TITLE_MAX = 40
+
+export function truncate(s: string, n: number): string {
+  return s.length > n ? `${s.slice(0, n - 1)}…` : s
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src/components/StatusBar/PrItem.tsx]: Fixed in 6dc0b8c — `truncate` now slices by code points via `Array.from()`, and a new surrogate-pair test pins the emoji-at-the-boundary case.

File: src-tauri/src/mod_engine/mods/git_monitor.rs [hunk 3]
```diff
@@ -245,12 +331,85 @@ async fn run_gh_pr(root: &str) -> Option<serde_json::Value> {
...
+            .or_else(|| item.get("state").and_then(|v| v.as_str()))
+            .unwrap_or("");
+        match s {
+            "SUCCESS" => c.passing += 1,
+            "FAILURE" | "ERROR" | "CANCELLED" | "TIMED_OUT" | "ACTION_REQUIRED" => {
+                c.failing += 1
+            }
+            "PENDING" | "IN_PROGRESS" | "QUEUED" | "WAITING" => c.pending += 1,
+            "SKIPPED" | "NEUTRAL" => c.skipped += 1,
+            _ => {}
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src-tauri/src/mod_engine/mods/git_monitor.rs]: Fixed in 6dc0b8c — `STARTUP_FAILURE` and `STALE` now count as failing (and `EXPECTED` as pending), with new Rust tests covering the silent-green-dot regression.

File: src-tauri/src/mod_engine/mods/git_monitor.rs [hunk 4]
```diff
@@ -245,12 +331,85 @@ async fn run_gh_pr(root: &str) -> Option<serde_json::Value> {
...
+
+    serde_json::json!({
+        "number": raw.get("number"),
+        "title": raw.get("title"),
+        "state": normalised_state,
+        "isDraft": raw.get("isDraft").and_then(|v| v.as_bool()).unwrap_or(false),
+        "url": raw.get("url"),
+        "checks": checks,
+    })
+}
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src-tauri/src/mod_engine/mods/git_monitor.rs]: Fixed in 6dc0b8c — `transform_pr` now returns `Option<Value>` and yields `None` when `number` / `title` / `url` are missing or wrong-typed, so the frontend never sees a malformed `PrInfo`.

File: src-tauri/src/mod_engine/mods/git_monitor.rs [hunk 5]
```diff
@@ -52,17 +84,40 @@ impl Mod for GitMonitorMod {
...
+                }
                 let cwd = cwd_rx.borrow().clone();
                 if let Some(cwd) = cwd {
                     let data = query_git_info(&cwd).await;
+                    let desired = desired_interval_secs(&data);
+                    if *interval_tx_for_timer.borrow() != desired {
+                        let _ = interval_tx_for_timer.send(desired);
+                    }
                     emitter.emit("git_monitor", "git_info", data);
                 }
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src-tauri/src/mod_engine/mods/git_monitor.rs:127]: Fixed in 6dc0b8c — switched the periodic-timer's interval read to `borrow_and_update()` so the watch's seen pointer advances on each read and cadence changes no longer cause an extra wake + `continue`.

File: src/components/StatusBar/StatusBarRight.tsx [hunk 6]
```diff
@@ -42,16 +43,21 @@ import {
...
+ *   └─────────────────────────────────────────────────────────────────────────────────────────────── process name
+ *
+ * The PR pill sits between git and cwd so the eye reads "branch → its PR →
+ * where I am on disk" — matches the existing left-to-right grouping logic.
+ * Pill is omitted when no PR is associated with the branch.
  *
  * Tab with no process data (shell tabs, or agent tabs before first poll):
  *
- *   status · ⎇ branch [●] [↑N] [↓N] · 📂 /cwd   (status hidden when idle)
+ *   status · ⎇ branch [●] [↑N] [↓N] · [●] ⊟ #123 title · 📂 /cwd   (status hidden when idle)
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src/components/StatusBar/StatusBarRight.tsx]: Fixed in 6dc0b8c — the no-process-data ASCII example now notes that the PR pill is omitted when the branch has no PR and the checks dot is hidden when no checks have reported yet.
